### ISSUE TITLE
Is there value in overriding: user_pref("browser.safebrowsing.downloads.remote.enabled", true); FOR LINUX USERS?

### ISSUE THREAD

**Issue by xe-3** [arkenfox/user.js#1812]
I use this override because, I followed the guidance:
> **_If you do not understand this_**, or you want this protection, then override this

Recently I was doing further research about SB in FF which gave me the impression that most/all of the SafeBrowsing _**download**_ protection is Windows (or Windows/MacOS) specific. I haven't seen this stated outright, but it seems implied by some of the language in the 'required reading' links for this override, as well as this SB test site: https://testsafebrowsing.appspot.com/

For a linux user is there any reason to override: `user_pref("browser.safebrowsing.downloads.remote.enabled", true)` or is this essentially a windows specific feature focused on windows specific filetypes?

I prefer to minimize the number of overrides I use, and I'm looking for a good reason to eliminate this override from my list.

**Comment by Thorin-Oakenpants** [comment]
right, so the https://github.com/arkenfox/user.js/wiki/3.2-Overrides-[Common] says to override `0403` if you're not sure what it means

This is, IMO, the only pref that sacrifices some security for more (possible) privacy. Personally, I don't think google would abuse it, as SB is used by 3+ billion devices and they have stated their position on it: someone, sometime, somehow would leak any abuse, it's just too big. But then stranger things have happened, and google are an advertising company (adSense etc) and the conflicts of interest are too great to ignore. And I also took into account that most users here, are fine with dealing with downloads - i.e getting shit from reputable/official sites, and having some AV (windows defender, whatever mac uses, and linux dudes will always do their thing .. terminals! :) ) ... so I honestly went against the grain, and disabled a safety feature. It's also telling that TB did this - they put privacy (because the outgoing connections can help with timing attacks, also fuck google) ahead of security here. SO it's not that big a deal to me, but I just wanted to make sure that it was front and center and listed. If you couldn;t follow it, then flip it to be safe - as a default choice. So I did my best

```
/* 0403: disable SB checks for downloads (remote)
 * To verify the safety of certain executable files, Firefox may submit some information about the
 * file, including the name, origin, size and a cryptographic hash of the contents, to the Google
 * Safe Browsing service which helps Firefox determine whether or not the file should be blocked
 * [SETUP-SECURITY] If you do not understand this, or you want this protection, then override this ***/
user_pref("browser.safebrowsing.downloads.remote.enabled", false);
```

So this is the only SB pref we set, and if you look at the links provided in section 0400, including the official info from firefox and a post from a moz developer who actually worked on this stuff back when it was set up (SBv4 and earlier), then all your answers are there (as well as the little blurb I wrote for section 0400). In fact, 0403 actually explains it (word for word IIRC from the firefox link)

I do not see how/why linux has anything to do with it - the pref and checks are in-app

What am I missing?

**Comment by Thorin-Oakenpants** [comment]
> I prefer to minimize the number of overrides I use, and I'm looking for a good reason to eliminate this override from my list.

don't override it. If you're responsible with your download sources (exe's dmgs, bats, scripts .. whatever is a `binary`) then you're fine. 95% of security is using your head

**Comment by Thorin-Oakenpants** [comment]
link [4] https://educatedguesswork.org/posts/safe-browsing-privacy/ is very good, it's by `eka` (Eric Rescorla: was chief something something at mozilla for like a decade - I met him in 2019 when the tor project team pitched the `super private browsing window mode` (basically FF with an extra know to open a tor window). I'm actually mentioned in that article .. see if you can find me

**Comment by xe-3** [comment]
> I do not see how/why linux has anything to do with it - the pref and checks are in-app

> What am I missing?

You are probably not missing anything. 

But let me briefly outline the context that led me to ask about Linux specifically:
1. I'm a longtime desktop Linux user, I'm very used to my OS being treated at best as an after thought, and often ignored completely.  :D
2. [This SB test page](https://testsafebrowsing.appspot.com/) seems to indicate most of the "Download" section is Windows specific or Windows/MacOS specific (I can't tell if that refers to the SB features or the test).
3. Also, all the tests in that section download a Windows specific file format (.exe) apart from one .zip
4. I couldn't find a list of software or filetypes on the blacklist, but I had some suspicion it would be mainly .exe files and things that are Windows specific or Windows/Mac specific.
5. Its uncommon to need to download software from the internet on Linux

So my question was frame as a Linux question not because I think Linux would be exempt or invulnerable, but because I am not sure if the Downloads component of SafeBrowsing actually provides any protection for Linux users, or if any common linux filetypes are even included on its blacklist. I am trying to find a list of what is blocked (or even just what filetypes) but I haven't found anything so far. I think I'll just follow your advice and remove the override, I'm pretty careful about what I download, and I never source software from the internet apart from git___ and, and/or direct from the dev or a trusted 3p.

While I have your attention on the topic of SafeBrowsing, have you heard any discussion from the Mozilla folks you communicate with about proxying connections to Google safe browsing servers? (I believe Safari and Brave both do this by default now). There is a [bug report (1516251)](https://bugzilla.mozilla.org/show_bug.cgi?id=1516251) asking for this feature but it didn't receive much traction.

**Comment by Thorin-Oakenpants** [comment]
ok, let me phrase it another way for you .. if you rarely download a `binary` in linux via the browser (and/or `binary` items are not a thing or less of a thing in linux) then flipping the pref back on or leaving it off as per AF is simply less of an issue. But the core tenet still holds. I did say `and linux dudes will always do their thing .. terminals! :)` but IANA very savvy linux user and I don't know much about it. I might poke the unicorns upstream at tor project, or corner one of them in Portugal over a ~~beer~~ pizza

The reason TB does not ship with SB ~~was two fold~~ is now legion. First, a new reason: over the years the relevance of the SB lists has dropped, you can actually get the numbers from google themselves and it's just way way way down on it's impact/hits (but that wasn't the case back then, just saying it is now). Another thing is we will likely ship uBO in Tor Browser (we just need to tweak it and it was discussed in some detail in [Sweden](https://gitlab.torproject.org/tpo/applications/team/-/wikis/Goteburg-2023-Meeting-Notes/ublock-origin) - I get around these days). But the main reason was not because we don't trust google (well, we don't .. fuck those guys, but I doubt they would ever compromise it or all hell would break loose), but because we need to control the traffic and something something timing attacks .. etc. It's easier to just kill the thing. TB choose simplicity and defense in depth over complexity

I doubt we will ever consider shipping SB with local files (it would bloat the sizes, not even sure we could do it from a license perspective). This is actually something we do with some Mozilla lists, we block the update but bundle the list at build time. Releases are every 4 weeks, good enough (for those lists, I forget what they are).

As for FF .. where is the threat here. So what, a SB local lists update check every 4? hours or whatever it is. Big deal. No PII. 180mn other FF users are dong the same thing every 4? hrs as well. Boo hoo.

If you are worried about your IP address, then use a fucking VPN :) JFC, are people thick. Brave and Safari's proxy is 99% privacy theater. Why should you trust Brave or Safari any more than Google.  So on that train of thought ... FF proxies it and now people will bitch about giving FF your IP address .. it never ends with this nonsense. People already spout crap here like #1807 - crap as in must be ZERO connections on startup

so tl;dr: not going to happen because the premise if flawed. There's nothing to do IMO (and rightly so, Mozilla have ignored it, if anything it might be a nice thing to do, but it's mostly marketing and feel good vibes for the masses). Besides, they have Mozilla VPN, you can use that.

**Comment by xe-3** [comment]
> Another thing is we will likely ship uBO in Tor Browser (we just need to tweak it and it was discussed in some detail in [Sweden](https://gitlab.torproject.org/tpo/applications/team/-/wikis/Goteburg-2023-Meeting-Notes/ublock-origin) - I get around these days).

This is (hopefully) great news! And I hope you enjoyed Sweden

I see what you are saying about the limited upside to proxying the connection to SB. Thanks for the perspective and guidance re: Safe Browsing, and thanks for all the work you put into AF & the Tor project. It is very much appreciated.

**Comment by Thorin-Oakenpants** [comment]
> JFC, are people thick?


![midvale](https://github.com/arkenfox/user.js/assets/16656956/f2f7070c-0eb4-4e09-9d17-3e6af574b701)

sorry, am going thru a farside renaissance period
